src/snyk/snykOss/services/vulnerabilityCount/npmModuleInfoFetchService.ts

```diff
@@ -0,0 +1,175 @@
+    if (!result) {
+      throw new Error(`Vulnerability result isn't available for ${key} due to missing cache entry.`);
+    }
+
+    return { ...module, vulnerabilityCount: result.totalVulns, tested: true };
+  }
+
+  private async getVulnerabilityCacheKey(module: ImportedModule): Promise<string | undefined> {
+    try {
+      const npmPackage = await this.getNpmRegistryPackage(module);
+      return `${npmPackage.name}@${npmPackage.version}`;
+    } catch (e) {
+      this.logger.error(e);
+      return undefined;
+    }
+  }
+
+  private async getNpmRegistryPackage(module: ImportedModule) {
+    let npmPackage: NpmRegistryPackage;
+    if (module.string) {
+      npmPackage = this.npmPackageCache.get(module.string) || (await this.fetchNpmRegistryPackage(module));
+      this.npmPackageCache.set(module.string, npmPackage);
+    } else {
+      npmPackage = await this.fetchNpmRegistryPackage(module);
+      this.npmPackageCache.set(module.string, npmPackage);
+    }
+
+    return npmPackage;
+  }
+
+  private async fetchNpmRegistryPackage(module: ImportedModule): Promise<NpmRegistryPackage> {
+    if (module.version && module.name) {
+      return { name: module.name, version: module.version };
+    }
+
+    let dir = this.projectCache.get(module.fileName);
+
+    if (!dir) {
+      const filename = finder(module.fileName).next().filename;
+      if (filename) {
+        dir = dirname(filename);
+        this.projectCache.set(module.fileName, dir);
+      } else {
+        throw new Error('Cannot find "package.json" to determine module key.');
+      }
+    }
+
+    const name = module.name;
+
+    const f = finder(join(dir, 'node_modules', name));
+    const moduleInfo = f.next().value;
+
+    // if the package doesn't start with the package name we were looking for
+    // then it means it's not locally installed, so let's get the version from the
+    // npm registry
+    if (!moduleInfo || !moduleInfo.name || !moduleInfo.name.toLowerCase().startsWith(name)) {
+      try {
+        const { data } = await npmRegistryApi.get<NpmRegistryPackage>(`${name}/latest`);
+
+        return { name, version: data.version };
+      } catch (err) {
+        return {
+          name,
+          version: 'latest',
+        };
+      }
+    }
+
+    return { name: moduleInfo.name, version: moduleInfo.version ?? 'latest' };
+  }
+
+  private lookupVulnerabilities(vulnerabilityKey: string, npmPackageCacheKey: string): Promise<NpmTestApiResult> {
+    const npmPackage = this.npmPackageCache.get(npmPackageCacheKey);
+    if (!npmPackage) {
+      throw new Error('Npm package is not in cache for some reason.');
+    }
+
+    return this.debouncePromise(vulnerabilityKey, this.test.bind(this), npmPackage);
+  }
+
+  private async test(npmPackage: NpmRegistryPackage): Promise<NpmTestApiResult> {
+    const utm = `utm_medium=${this.config.source}&utm_source=${this.config.source}&utm_campaign=${this.config.source}`;
```

Comment:
All three are config.source? Would it be possible to either have different values or alternatively remove redundant information?

Reply:
To my understanding all three params are required. They have the same value similarly to authentication integration via `code-client` https://github.com/snyk/code-client/blob/0c59c18440bb77c3c51eb620bba0ff3e1b324ad8/src/http.ts#L107. Vuln Cost had the following:
`utm_medium=ide&utm_source=vscode&utm_campaign=vuln_cost`. We don't have a campaign for it, so `vscode` will be there instead, and `utm_medium=ide` seems wrong to me, because we don't send this parameter set to 'ide' from any of our plugins.